From 8c324bed4ea0bf1f6def6275b13cfff0ffc9c177 Mon Sep 17 00:00:00 2001
From: Prakash Brahmajyosyula <bprakash@marvell.com>
Date: Mon, 3 Dec 2018 11:33:52 +0300
Subject: [PATCH 0843/1051] soc: octeontx2: cleanup, fix for possible memory
 leak.

strsep modifies cmd_buf base addr and freeing modified cmd_buf may lead
memory leak or crash. With this patch parsing preserved base addr of
cmd_buf to kfree.

Change-Id: I6618cee8763524192ce4f2caabdc51b0df42f323
Signed-off-by: Prakash Brahmajyosyula <bprakash@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/1751
Tested-by: sa_ip-sw-jenkins
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/soc/marvell/octeontx2/rvu_debugfs.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/soc/marvell/octeontx2/rvu_debugfs.c b/drivers/soc/marvell/octeontx2/rvu_debugfs.c
index 6f8c06b11c93..264ada698e29 100644
--- a/drivers/soc/marvell/octeontx2/rvu_debugfs.c
+++ b/drivers/soc/marvell/octeontx2/rvu_debugfs.c
@@ -202,6 +202,7 @@ static ssize_t rvu_dbg_npa_qsize_display(struct file *filp,
 		*cmd_buf_tmp = '\0';
 		count = cmd_buf_tmp - cmd_buf + 1;
 	}
+	cmd_buf_tmp = cmd_buf;
 	subtoken = strsep(&cmd_buf, " ");
 	ret = subtoken ? kstrtoint(subtoken, 10, &npalf) : -EINVAL;
 	if (cmd_buf)
@@ -251,7 +252,7 @@ static ssize_t rvu_dbg_npa_qsize_display(struct file *filp,
 	}
 	kfree(buf);
 npa_qsize_display_done:
-	kfree(cmd_buf);
+	kfree(cmd_buf_tmp);
 	return count;
 }
 
-- 
2.17.1

